common/types/endpoint.go

```diff
@@ -138,6 +142,60 @@ type Endpoint struct {
 	Consumable       *Consumable           `json:"consumable"`
 	PolicyMap        *policymap.PolicyMap  `json:"-"`
 	Opts             *BoolOptions          `json:"options"` // Endpoint bpf options.
+	Status           EndpointStatus        `json:"status"`
+}
+
+type statusLog struct {
+	Status    Status    `json:"status"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+type EndpointStatus struct {
+	Log     *[maxLogs]*statusLog `json:"log"`
+	Index   uint8                `json:"index"`
```

Comment:
I know, I did that because I didn't want to write the following code eheh:
```
func (e *Endpoint) LogStatus(code StatusCode, msg string) {
....
      e.Status.Index++
      if e.Status.Index >= maxLogs {
         e.Status.Index = 0
      }
```
I'll make that change ;)